app/src/main/kotlin/github/com/rhacco/dota2androidapp/viewmodel/MatchesViewModel.kt

```diff
@@ -6,49 +6,85 @@ import android.arch.lifecycle.MediatorLiveData
                         { error ->
                             mIsLoadingLiveData.value = false
                             Log.d(App.instance.getString(R.string.log_msg_debug),
-                                    "Failed to update top live games: " + error)
+                                    "Failed to fetch top live games: " + error)
                         }
                 ))
     }
 
-    fun getRealtimeStats(averageMMR: Int, serverSteamId: Long) {
+    fun getLiveMatchListData(averageMMR: Int, serverSteamId: Long) {
         mIsLoadingLiveData.value = true
         mDisposables.add(RealtimeStatsRepository.getRealtimeStats(serverSteamId)
                 .subscribe(
                         { result ->
                             mIsLoadingLiveData.value = false
-                            mRealtimeStatsQueryLiveData.value = Pair(averageMMR, result)
+                            val listItemData: LiveMatchesListAdapter.ListItemData =
+                                    LiveMatchesListAdapter.ListItemData()
+                            listItemData.mAverageMMR = averageMMR
+                            if (averageMMR < 1)
+                                listItemData.mTitle = "Tournament Match (Match ID " +
+                                        result.match.matchid + ")"
+                            else
+                                listItemData.mTitle = "Ranked Match (Average MMR " + averageMMR +
+                                        ", Match ID " + result.match.matchid + ")"
+                            if (result.teams.size == 2) {
+                                result.teams[0].players
+                                        .asSequence()
+                                        .map {
+                                            if (it.name.length > 20)
+                                                it.name.substring(0, 16) + "..."
+                                            else
+                                                it.name
+                                        }
+                                        .forEach {
+                                            listItemData.mPlayersRadiant =
+                                                    listItemData.mPlayersRadiant + "\n" + it
+                                        }
+                                result.teams[1].players
+                                        .asSequence()
+                                        .map {
+                                            if (it.name.length > 20)
+                                                it.name.substring(0, 16) + "..."
+                                            else
+                                                it.name
+                                        }
+                                        .forEach {
+                                            listItemData.mPlayersDire =
+                                                    listItemData.mPlayersDire + "\n" + it
```

Comment:
Ew. Please see my comment on the layout about removing this. You absolutely do not need to do this \n

Regardless if you choose to use databinding or not, please do this sort of binding/formatting in the adapter.

Reply:
Fixed